dataRetriever/requestHandlers/baseRequestHandler.go

```diff
@@ -0,0 +1,62 @@
+package requestHandlers
+
+import (
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/elrond-go-core/marshal"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+)
+
+// ArgBaseRequestHandler is the argument structure used as base to create a new a request handler instance
+type ArgBaseRequestHandler struct {
+	SenderResolver dataRetriever.TopicResolverSender
+	Marshaller     marshal.Marshalizer
+}
+
+type baseRequestHandler struct {
```

Comment:
I would call it something like baseRequester as we already have RequestHandler implementation & interface definition

Reply:
done